src/bios.rs

```diff
@@ -142,10 +144,63 @@ impl Component for Bios {
         crate::component::query_adopt_state()
     }
 
+    // Backup the current grub.cfg and install new static config
+    // - Backup "/boot/loader/grub.cfg" to "/boot/grub2/grub.cfg.bak"
+    // - Remove symlink "/boot/grub2/grub.cfg"
+    // - Install new static grub.cfg
+    // - Remove old "/boot/loader/grub.cfg"
+    fn migrate_static_grub_config(&self, sysroot_path: &str, destdir: &openat::Dir) -> Result<()> {
+        use anyhow::Context;
+        use openat_ext::OpenatDirExt;
+
+        // sysroot_path is /, destdir is Dir of /
+        let grub_config_path = Utf8PathBuf::from(sysroot_path).join("boot/grub2");
+        let grub_config_dir = destdir
+            .sub_dir("boot/grub2")
+            .context("Opening boot/grub2")?;
+
+        let grub_config = grub_config_path.join(grubconfigs::GRUBCONFIG);
+
+        // On BIOS, /boot/grub2/grub.cfg is symlink to /boot/loader/grub.cfg,
+        // backup it to /boot/grub2/grub.cfg.bak
+        if !grub_config.exists() {
+            anyhow::bail!("Not found '{}'", grub_config);
+        } else if !grub_config.is_symlink() {
+            anyhow::bail!("'{}' is not a symlink", grub_config);
```

Comment:
Should this be an error? If this is not a symlink then we need to keep going as the migration might have been interrupted after we did symlink swap.

Reply:
If `/boot/grub2/grub.cfg` is not symlink, maybe skip backup, then will install / overwrite the `grub.cfg` with static config, and can not remove the config as it is the static config that we need.
What we want to remove is `/boot/grub2/../loader/grub.cfg`, but now it is `/boot/grub2/grub.cfg`. Need to check if it is symlink before `remove_file`, WDYT?

Reply:
https://github.com/coreos/bootupd/pull/945#discussion_r2156982205